test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -0,0 +1,388 @@
+        phaseEvent.SetEvent();
+    }
+}
+
+void initUnitTestMapping()
+{
+    switchMapping["ChannelRequestUsingNullRemoteId"] = UnitTest::channelRequestUsingNullRemoteId;
+    switchMapping["ChannelRequestUsingRemoteId"] = UnitTest::channelRequestUsingRemoteId;
+    switchMapping["MultipleChannelRequestUsingSameRemoteId"] = UnitTest::multipleChannelRequestUsingSameRemoteId;
+    switchMapping["MultipleChannelRequestUsingMultipleRemoteId"] = UnitTest::multipleChannelRequestUsingMultipleRemoteId;
+    switchMapping["ThreeChannelRequestUsingSameRemoteId"] = UnitTest::threeChannelRequestUsingSameRemoteId;
+    switchMapping["RegisterActivator"] = UnitTest::registerActivator;
+    switchMapping["UnregisterActivator"] = UnitTest::unregisterActivator;
+}
+
+bool ChannelRequestUsingNullRemoteId()
+{
+    winrt::hresult hr = S_OK;
+
+    try
+    {
+        auto channelOperation = PushNotificationManager::CreateChannelAsync(GUID_NULL).get();
+    }
+    catch (...)
+    {
+        auto channelRequestException = hresult_error(to_hresult());
+        hr = channelRequestException.code();
+    }
+
+    return (hr == E_INVALIDARG);
+}
+
+bool ChannelRequestUsingRemoteId()
+{
+    wil::unique_handle channelEvent = wil::unique_handle(CreateEvent(nullptr, FALSE, FALSE, nullptr));
+    auto hr = S_OK;
+    auto channelOperation = PushNotificationManager::CreateChannelAsync(remoteId1);
+
+    channelOperation.Progress(
+        [](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& /* sender */,
+            PushNotificationCreateChannelStatus const& args)
+        {
+            if (args.status == PushNotificationChannelStatus::InProgress)
+            {
+            }
+            else if (args.status == PushNotificationChannelStatus::InProgressRetry)
+            {
+            }
+        });
+
+    // Setup the completed event handler
+    channelOperation.Completed(
+        [&channelEvent](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& sender,
+            AsyncStatus const /* asyncStatus */)
+        {
+            auto result = sender.GetResults();
+            if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+            {
+                auto channelUri = result.Channel().Uri();
+                auto channelExpiry = result.Channel().ExpirationTime();
+            }
+            else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+            {
+            }
+
+            SetEvent(channelEvent.get());
+        });
+
+    // The maximum amount of time it takes for channel request to be obtained - 16mins
+    if (WAIT_OBJECT_0 != WaitForSingleObject(channelEvent.get(), 960000 /* milliseconds */))
+    {
+        channelOperation.Cancel();
+        hr = ERROR_TIMEOUT;
+    }
+    else
+    {
+        channelOperation.Close(); // Do not call getresults after this
```

Comment:
What happens if Close() is not called?

Reply:
nothing is going to happen. Just a good practice and that is why I added an extra comment since I am calling close on what not to do after that.